common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -172,7 +178,9 @@ protected void serviceInit(Configuration conf) throws Exception {
       }
 
       TransportConf transportConf = new TransportConf("shuffle", new HadoopConfigProvider(conf));
-      blockHandler = new ExternalBlockHandler(transportConf, registeredExecutorFile);
+      shuffleMergeManager = new RemoteBlockPushResolver(transportConf, APP_BASE_RELATIVE_PATH);
```

Comment:
Shall we make this configurable?

Reply:
On the client side, we have added a configuration `spark.shuffle.push.enabled`. This is per application and by default it is going to be `false`. An app would decide whether it wants to run with push enabled or not. 
If we make this configurable on the server-side, how would we enforce that an application doesn't try push-based shuffle when the servers don't support it?
Also, I think changing server side configurations is a bit of hassle. We run Spark over Yarn and for every server side change we need restart all the NodeManagers.
Do you have any concerns that the `shuffleMergeManager` interferes with regular shuffle?

Reply:
yeah I think that is the question - are there any side affects that I don't want if I'm not using it. we are registering applications and executors with it even if not in use. I assume that will at least use a bit more memory.  Personally I think its safer to have a way to totally turn it off, I think we could leave that config not published, though really I think all of them are that way.  This also allows someone to turn it off in case they don't want clients to be able to do it at all.  I was also wondering if someone might want to have their own custom implementation for his. So if we made it configurable such that they could provide a class, that would also be supported.

Reply:
The only overhead is the `appsPathInfo` map maintained in `RemoteBlockPushResolver`, which is a per application map recording the local dir paths used for storing merged shuffle partition files for this application on a given node.
Since it's at application level, the additional memory footprint is very negligible when push-based shuffle is disabled.

Would it be OK if we introduce another server side config as a cluster-admin managed config as a master switch to enable/disable push-based shuffle?
If disabled, we can just use the `NoOpMergedShuffleFileManager` from #29855 here instead.

Reply:
@Ngone51 , @tgravescs , @Victsm 
I will add a configuration on the server side that will disable push-based shuffle on the server.

@tgravescs Do you think the custom implementation can be a follow-up task? I have create this jira https://issues.apache.org/jira/browse/SPARK-33235 for all the follow-up tasks. I will add the custom implementation as a subtask as well.

Reply:
I was just thinking it would be easier to create the config now that is a class that is either the push based or the Noop one. Then a user could also just specify a different class that is their custom one.  I'm fine with doing that separate.

Reply:
Yes, doesn't seem to be a big change to have a config that is a class. I will add that with this change. 
@tgravescs

Reply:
+1 to add config on the server-side, you can leave the default value to `RemoteBlockPushResolver`. Thanks @otterc

Reply:
I have added this configuration cc. @tgravescs @Ngone51